app/components/primer/alpha/x_banner.ts

```diff
@@ -1,8 +1,35 @@
 import {controller, target} from '@github/catalyst'
+// import {announceFromElement} from '@primer/live-region-element'
 
 @controller
 class XBannerElement extends HTMLElement {
   @target titleText: HTMLElement
+  @target banner: HTMLElement
+  private observer?: IntersectionObserver
+
+  connectedCallback() {
+    if (this.#announceOnShow === 'true') {
+      const options = {
+        root: document.body,
+        rootMargin: '0px',
+        threshold: 0,
+      }
+      const callback = (entries: IntersectionObserverEntry[]) => {
+        for (const entry of entries) {
+          if (entry.isIntersecting || entry.intersectionRatio > 0) {
+            // Uncomment when available: https://github.com/primer/live-region-element/
+            // announceFromElement(this.banner)
+            this.observer?.disconnect()
+          }
+        }
+      }
+      this.observer = new IntersectionObserver(callback, options)
```

Comment:
Looked into this because it was mentioned in https://github.com/primer/view_components/pull/2576#issuecomment-1933983766, but I dont know if its handling all the scenarios and if it'll work the way we expect yet.

Needs further testing!